src/main/java/duke/component/Parser.java

```diff
@@ -0,0 +1,138 @@
+package duke.component;
+
+//imports
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyDescException;
+import duke.exception.InvalidArgumentException;
+
+/**
+ * Represents a Parser object that parses user inputs.
+ * The 'Parser' class supports operators (i) parse out user inputs
+ * and (ii) determine the input type, which helps parse out user inputs.
+ */
+public class Parser {
+
+
+    /**
+     * Initialises a new Parser object.
+     */
+    public Parser() {
+    }
+
+
+    /**
+     * Parses out user inputs.
+     *
+     * @param input User input.
+     * @return Command Type of command that user input in.
+     * @throws DukeException If there is IOException when reading or writing from text file.
+     */
+    public static Command parse(String input) throws DukeException {
+        switch (input) {
+
+        case "bye":
+            return new ExitCommand();
+
+
+        case "list":
+            return new ListCommand();
+
+
+        default:
+            return determineInputType(input);
+
+        }
+    }
+
+    /**
+     * Returns Command object based on user input.
+     * Helper method to parse(String input) method.
+     *
+     * @param input User input.
+     * @return Command Type o comand summoned by user.
+     * @throws DukeException If there is IOException when reading or writing from text file.
+     */
+    public static Command determineInputType(String input) throws DukeException {
+        if (input.contains("done")) {
+
```

Comment:
I would be careful with using input.contains() because for cases where the input is, for example:
`todo select deadline for goals`
Or conversely: 
`deadline do up a todo list /by 12/12/19 12:30`
Your code might then not work as intended. You can consider instead using the startsWith() function (https://www.tutorialspoint.com/java/java_string_startswith.htm). 😃